crates/astria-sequencer/src/app.rs

```diff
@@ -315,10 +319,42 @@ impl App {
             .try_into()
             .map_err(|_| anyhow!("chain IDs commitment must be 32 bytes"))?;
 
-        let signed_txs = self
-            .execute_block_data(txs)
+        let expected_txs_len = txs.len();
+
+        let block_data = BlockData {
+            txs: txs.into_iter().collect(),
+            misbehavior: process_proposal.misbehavior,
+            height: process_proposal.height,
+            time: process_proposal.time,
+            next_validators_hash: process_proposal.next_validators_hash,
+            proposer_address: process_proposal.proposer_address,
+        };
+
+        let txs = self
+            .pre_execute_transactions(block_data)
             .await
-            .context("transactions failed to decode and execute")?;
+            .context("failed to prepare for executing block")?;
+
+        // we don't care about the cometbft max_tx_bytes here, as cometbft would have
+        // rejected the proposal if it was too large.
+        // however, we should still validate the other constraints, namely
+        // the max sequenced data bytes.
+        let mut block_size_constraints = BlockSizeConstraints::new(usize::MAX)
```

Comment:
Maybe add a new constructor `BlockSizeConstraints::unlimited_cometbft` or something like that?